### ISSUE TITLE
Rule idea: ScopedViewComponentLeak - Detect memory leaks from scoped UI components

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#29]
## Problem

Injecting UI-lifecycle-dependent objects (RecyclerView.Adapter, View) using fragment scope causes memory leaks because scoped instances hold view references after `onDestroyView`.

## Related Koin Issues

- https://github.com/InsertKoinIO/koin/issues/1144
- https://github.com/InsertKoinIO/koin/issues/1122
- https://github.com/InsertKoinIO/koin/issues/149

## Detection Pattern

```kotlin
// ❌ Bad: scoped Adapter holds view references
scope<MyFragment> {
    scoped { MyAdapter() }
}

// ✅ Good: factory creates new instance per injection
scope<MyFragment> {
    factory { MyAdapter() }
}
```

## Implementation Challenge

Requires Android platform type detection (RecyclerView.Adapter, View subclasses).

## Priority

LOW - Consider for v0.6.0

## Category

`scope` / `platform`

**Comment by kirich1409** [comment]
**Status: deferred — implementable, low priority.**

This rule is technically feasible with PSI heuristics: detect when a ViewModel-scoped dependency is held as a property in an Activity or Fragment that outlives the ViewModel's scope. However, the detection requires reliable type hierarchy analysis combined with scope tracking — the same heuristic approach already used in `ScopeDeclareWithActivityOrFragment`.

Leaving open for a future iteration. A PR with tests is welcome.
